### PR TITLE
Create a version catcher

### PR DESCRIPTION
**Pull request by McKnight-42**
Create/augment existing gha to check latest version of dbt-core to update scaffold of project structure and keep adapter plugin start point within step bounds of core.

co-author @kwigley

### CODE REVIEW

**Comment by kwigley** [comment]
@McKnight-42 can you change the base branch of this to make it easier to see the diff?

**Comment by McKnight-42** [comment]
@kwigley great catch.

File: .github/workflows/version.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: Nightly dbt-version check and update.
+on:
+  push:
```
Comment (kwigley) [inline .github/workflows/version.yml:3]:
```suggestion
```
I think we can just run this nightly and when manually triggered, no need to run on every push. This was left over from testing. I just figured out we can test this action locally with `act -W .github/workflows/version.yml --detect-event`. Can you add a comment to the top of the file mentioning that you can test this workflow locally with `act` using that command?

File: .github/workflows/version.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+name: Nightly dbt-version check and update.
```
Comment (kwigley) [inline .github/workflows/version.yml:1]:
```suggestion
name: Check and update dbt version
```
nit

File: .github/workflows/version.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  workflow_dispatch:
+  schedule:
+    - cron: "0 5 * * *" # 5 UTC
+
+jobs:
+  parse:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get Package Info
```
Comment (kwigley) [inline .github/workflows/version.yml:13]:
```suggestion
      - name: Get Latest Package Info for dbt-core
```

File: .github/workflows/version.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        id: package-info
+        uses: dbt-labs/actions/py-package-info@v1
+        with:
+          package: "dbt-core"
+
+      - name: Use Package Info
```
Comment (kwigley) [inline .github/workflows/version.yml:19]:
```suggestion
      - name: dbt-core Version
```

File: .github/workflows/version.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        run: |
+          echo version: ${{ steps.package-info.outputs.version }}
+
+      - name: Parse Semver
+        id: parse-valid
+        uses: dbt-labs/actions/parse-semver@v1
+        with:
+          version: ${{ steps.package-info.outputs.version }}
+
+      - name: Use Parsed Output
```
Comment (kwigley) [inline .github/workflows/version.yml:29]:
```suggestion
      - name: Parsed dbt-core Version
```

**Review by kwigley — CHANGES_REQUESTED** [review]
awesome, I'm super happy we got to reuse actions from https://github.com/dbt-labs/actions

File: .github/workflows/version.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
+name: Nightly dbt-version check and update.
+on:
+  push:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 5 * * *" # 5 UTC
+
+jobs:
+  parse:
```
Comment (kwigley) [inline .github/workflows/version.yml:9]:
```suggestion
  update-version:
```

File: cookiecutter.json [hunk 7]
```diff
@@ -1,18 +1,21 @@
...
-  "is_sql_adapter": ["true", "false"],
-  "project_version":"1.0.0",
-  "dbt_version":"1.0.0",
-  "package_name":"dbt-{{ cookiecutter.directory_name }}",
+  "is_sql_adapter": [
+    "true",
+    "false"
+  ],
+  "project_version": "1.0.0",
+  "dbt_version": "1.0.4",
```
Comment (kwigley) [inline cookiecutter.json:14]: 👏 👏

File: .github/workflows/version.yml (hunk 1 above)
Comment (McKnight-42) [reply .github/workflows/version.yml]: removed and added comments

File: .github/workflows/version.yml (hunk 3 above)
Comment (McKnight-42) [reply .github/workflows/version.yml]: suggested change made

File: .github/workflows/version.yml (hunk 4 above)
Comment (McKnight-42) [reply .github/workflows/version.yml]: suggested change made

File: .github/workflows/version.yml (hunk 5 above)
Comment (McKnight-42) [reply .github/workflows/version.yml]: suggested change made

File: .github/workflows/version.yml (hunk 6 above)
Comment (McKnight-42) [reply .github/workflows/version.yml]: suggested change made

File: .github/workflows/version.yml (hunk 2 above)
Comment (McKnight-42) [reply .github/workflows/version.yml]: suggested change made

**Review by ChenyuLInx — APPROVED** [review]
This looks good to me! How do we test GHA before merging it in now?

**Comment by McKnight-42** [comment]
wanted to move https://github.com/dbt-labs/dbt-database-adapter-scaffold/pull/24#issuecomment-1100389318 over to here as its still a open topic.

File: .github/workflows/version.yml [hunk 8]
```diff
@@ -0,0 +1,73 @@
+# to test locally please add push to the on: workflow selection, normally we will only want to
+# test this nightly so on push isn't needed for that but is for using act locally to later run and check.
```
Comment (kwigley) [inline .github/workflows/version.yml:2]:
```suggestion
# To test locally with `act` (https://github.com/nektos/act), run `act -W .github/workflows/version.yml --detect-event`
```

no need to add the `push` trigger to test! you can use the above command to test locally without any changes

**Review by kwigley — APPROVED** [review]
LGTM 👍

**Pull request merged**